plugins/available/less-pretty-cat.plugin.bash

```diff
@@ -1,31 +1,31 @@
+# shellcheck shell=bash
 cite about-plugin
 about-plugin 'pygmentize instead of cat to terminal if possible'
 
-if $(command -v pygmentize &> /dev/null) ; then
-  # get the full paths to binaries
-  CAT_BIN=$(which cat)
-  LESS_BIN=$(which less)
-  BASH_IT_CCAT_STYLE="${BASH_IT_CCAT_STYLE:=default}"
-  BASH_IT_CLESS_STYLE="${BASH_IT_CLESS_STYLE:=default}"
+_command_exists pygmentize || return
 
-  # pigmentize cat and less outputs - call them ccat and cless to avoid that
-  # especially cat'ed output in scripts gets mangled with pygemtized meta characters
-  function ccat()
-  {
-      about 'runs either pygmentize or cat on each file passed in'
-      param '*: files to concatenate (as normally passed to cat)'
-      example 'cat mysite/manage.py dir/text-file.txt'
-      for var;
-      do
-          pygmentize -f 256 -O style="$BASH_IT_CCAT_STYLE" -g "$var" 2>/dev/null || "$CAT_BIN" "$var";
-      done
-  }
+# pigmentize cat and less outputs - call them ccat and cless to avoid that
+# especially cat'ed output in scripts gets mangled with pygemtized meta characters
+function ccat() {
+	about 'runs either pygmentize or cat on each file passed in'
+	param '*: files to concatenate (as normally passed to cat)'
+	example 'cat mysite/manage.py dir/text-file.txt'
 
-  function cless()
-  {
-      about 'it pigments the file passed in and passes it to less for pagination'
-      param '$1: the file to paginate with less'
-      example 'less mysite/manage.py'
-      pygmentize -f 256 -O style="$BASH_IT_CLESS_STYLE" -g $* | "$LESS_BIN" -R
-  }
-fi
+	local file
+	: "${BASH_IT_CCAT_STYLE:=default}"
+
+	for file in "$@"; do
+		pygmentize -f 256 -O style="$BASH_IT_CCAT_STYLE" -g "$file" 2> /dev/null || command cat "$file"
```

Comment:
I see the goal here, but *how* could `pygmetize` fail in a way that would defer to `cat` ?

Even if it couldn't offer any colorization, seems it would still act as a standard `cat` in that case?

The `cless` function assumes no failures, which further enforces the idea.

Reply:
No idea; I've removed that bit now